File: proy_client/src/pages/Fevolution/Fevolution.js
import { PlusOutlined } from '@ant-design/icons';
import {
    Button,
    Col,
    DatePicker,
    Form,
    Input,
    Row,
    Select,
} from 'antd';
import { useState } from 'react';
import './Fevolution.scss';

const { RangePicker } = DatePicker;
const { TextArea } = Input;

const normFile = (e) => {
    if (Array.isArray(e)) {
        return e;
    }
    return e?.fileList;
};

export const Fevolution = () => {
    const [componentDisabled, setComponentDisabled] = useState(true);
    return (
        <div className='container'>
            <Form
            labelCol={{
            span: 10,
            }}
            wrapperCol={{
            span: 14,
            }}
            layout="horizontal"
            //disabled={componentDisabled}
            style={{
            maxWidth: 600,
            }}
            >
                <h5 className='label-fecha'>Fecha: </h5>
                <h5 className='label-fecha'>Hora: </h5>
                <Row>
                    <Col >
                        
                    </Col>
                </Row>
                <Form.Item label="Diagnóstico médico">
                            <Select placeholder="Info api C01X">
                                <Select.Option value="demo">api</Select.Option>
                            </Select>
                        </Form.Item>
                <Form.Item label="Nombre Completo">
                    <Input />
                </Form.Item>
                <Form.Item label="Documento">
                    <Input />
                </Form.Item>
                <Form.Item label="Fecha de nacimiento">
                    <Input />
                </Form.Item>
                <Form.Item label="Procedencia">
                    <Input />
                </Form.Item>
                <Form.Item label="EPS">
                    <Input />
                </Form.Item>
                <Form.Item label="Dirección de residencia">
                    <Input />
                </Form.Item>
                <Form.Item label="Teléfono/Celular">
                    <Input />
                </Form.Item>
                <Form.Item label="Posicioón de juego">
                    <Input />
                </Form.Item>
                <Form.Item label="Tiempo de vinculación">
                    <Input />
                </Form.Item>
                <Form.Item label="Club de procedencia">
                    <Input />
                </Form.Item>
                <Form.Item label="Ejercicio físico adicional">
                    <Input />
                </Form.Item>
                <Form.Item label="Interpretación/Observación">
                    <TextArea rows={4} />
                </Form.Item>
                <Form.Item label="Button">
                <Button>Salir</Button>
                </Form.Item>
            </Form>
        </div>
        
    );
};

//export default () => <FormDisabledDemo />;

File: server-project/src/models/fevolution.py
class Evolution:
    def __init__(self, date, time, medical_diagnosis, full_name, document, date_birth, origin
                , eps, address, phone, position, time_linkage, club_home, additional_exercise
                , gender, day, activity_type, observation, approved):
        ''' 
        date = fecha
        time = hora
        medical_diagnosis = diagnóstico médico
        full_name = nombre completo
        document = documento
        date_birth = fecha de nacimiento
        origin = procedencia
        eps = eps
        address = direccion de residencia
        phone = telefono
        position = posición de juego
        time_linkage = tiempo de vinculación
        club_home = club de procedencia
        additional_exercise = ejercicio físico adicional
        gender = género
        day = jornada
        activity_type = tipo de actividad realizada
        observation = observación / interpretación
        approved = aprobado
        '''
        self.date = date
        self.time = time
        self.medical_diagnosis = medical_diagnosis
        self.full_name = full_name
        self.document = document
        self.date_birth = date_birth
        self.origin = origin
        self.eps = eps
        self.address = address
        self.phone = phone
        self.position = position
        self.time_linkage = time_linkage
        self.club_home = club_home
        self.additional_exercise = additional_exercise
        self.gender = gender
        self.day = day
        self.activity_type = activity_type
        self.observation = observation
        self.approved = approved

    def toDBCollection(self):
        return{
            "date":self.date,
            "time":self.time,
            "medical_diagnosis":self.medical_diagnosis,
            "full_name":self.full_name,
            "document":self.document,
            "date_birth":self.date_birth,
            "origin":self.origin,
            "eps":self.eps,
            "address":self.address,
            "phone":self.phone,
            "position":self.position,
            "time_linkage":self.time_linkage,
            "club_home":self.club_home,
            "additional_exercise":self.additional_exercise,
            "gender":self.gender,
            "day":self.day,
            "activity_type":self.activity_type,
            "observation":self.observation,
            "approved":self.approved
        }
